File: src/030. Digit fifth powers/030.py
def sum_of_fifth_power(num):
    sum = 0
    while num > 0:
        sum += (num % 10) ** 5
        num /= 10
    return sum

result = []
for num in range(2, 1000000):
    if sum_of_fifth_power(num) == num:
        result.append(num)

print(sum(result))

File: src/030. Digit fifth powers/030.cpp
#include <iostream>

static int digit_fifth_power_sum(int n);

int main() {
    int sum = 0;
    for (int i = 2; i < 1000000; i++) {
        if (digit_fifth_power_sum(i) == i) {
            sum += i;
        }
    }
    std::cout << sum << std::endl;
}

static int digit_fifth_power_sum(int n) {
    int sum = 0;
    while (n > 0) {
        int d = n % 10;
        sum += d * d * d * d * d;
        n /= 10;
    }
    return sum;
}
